### PR TITLE
MM-36721 [GEKIDOU] Read Messages in Channel - Channel Title Component

### PR DESCRIPTION
**Pull request by avinashlng1080**
#### Summary
As part of the 'Read Messages In Channel' feature, this PR contains only the porting of the ChannelNavBar component.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-36721

### CODE REVIEW

File: app/init/launch.ts [hunk 1]
```diff
@@ -51,11 +53,11 @@ const launchApp = async (props: LaunchProps, resetNavigation = true) => {
             break;
         }
     }
-
     serverUrl = await getActiveServerUrl();
 
     if (serverUrl) {
         const credentials = await getServerCredentials(serverUrl);
+        await setupChannelMockData();
```
Comment (avinashlng1080) [inline app/init/launch.ts:60]: Do not commit this file.

File: tsconfig.json [hunk 2]
```diff
@@ -37,6 +37,7 @@
             "@app/*": ["app/*"],
             "@assets/*": ["dist/assets/*"],
             "@client/*": ["app/client/*"],
+            "@contexts/*": ["app/contexts/*"],
```
Comment (avinashlng1080) [inline tsconfig.json:40]: Do not commit this new context alias

File: test/setup.ts [hunk 3]
```diff
@@ -7,8 +7,9 @@ import * as ReactNative from 'react-native';
```
Comment (avinashlng1080) [inline test/setup.ts:7]: Don't commit file

File: test/mock_database_data.ts [hunk 4]
```diff
@@ -0,0 +1,227 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
```
Comment (avinashlng1080) [inline test/mock_database_data.ts:2]: Do not commit file

File: index.ts [hunk 5]
```diff
@@ -10,12 +10,13 @@ import DatabaseManager from './app/database/manager';
 import {getAllServerCredentials} from './app/init/credentials';
 import GlobalEventHandler from './app/init/global_event_handler';
 import {initialLaunch} from './app/init/launch';
-import NetworkManager from './app/init/network_manager';
 import ManagedApp from './app/init/managed_app';
+import NetworkManager from './app/init/network_manager';
```
Comment (avinashlng1080) [inline index.ts:14]: Do not commit file

File: babel.config.js [hunk 6]
```diff
@@ -23,13 +23,15 @@ module.exports = {
                 '@app': './app/',
                 '@assets': './dist/assets',
                 '@client': './app/client',
+                '@contexts': './app/contexts',
```
Comment (avinashlng1080) [inline babel.config.js:26]: Undo

File: app/utils/user/index.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    if (ids[0] === userId) {
+        otherUserId = ids[1];
+    } else {
+        otherUserId = ids[0];
+    }
+
+    return otherUserId;
+}
+
+export function isInRole(roles: string, inRole: string) {
```
Comment (avinashlng1080) [inline app/utils/user/index.ts:29]: validate the usage of utils methods with @enahum

**Comment by enahum** [comment]
@avinashlng1080 do we keep this PR opened ?

**Comment by avinashlng1080** [comment]
Closing this PR.  It is no more needed.

**Pull request closed without merging**